pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -74,21 +76,51 @@ func (s *spdxParser) getTopLevelPackage() error {
 	// TODO: Add CertifyPkg to make a connection from GUAC purl to OCI purl guessed
 	// oci purl: pkg:oci/debian@sha256%3A244fd47e07d10?repository_url=ghcr.io/debian&tag=bullseye
 
-	// TODO (dejanb): do a quick check for SPDX to see if there are any "DESCRIBES" relationships
-	// from the document and omit this top level heuristic package otherwise.
+	var spdxIds []string
+	for _, r := range s.spdxDoc.Relationships {
+		// If both sides of the relationship contain the same string,
+		// it is not a valid DESCRIBES/DESCRIBED_BY relationship.
+		if r.RefA.ElementRefID == r.RefB.ElementRefID {
+			continue
+		}
+
+		if r.RefA.ElementRefID == "DOCUMENT" && r.Relationship == spdx_common.TypeRelationshipDescribe {
+			spdxIds = append(spdxIds, string(r.RefB.ElementRefID))
+		} else if r.Relationship == spdx_common.TypeRelationshipDescribeBy && r.RefB.ElementRefID == "DOCUMENT" {
+			spdxIds = append(spdxIds, string(r.RefA.ElementRefID))
+		}
+	}
 
-	// Currently create TopLevel package as well in some cases where we guess that the SPDX document
-	// may not encode it
-	var purl string = "pkg:guac/spdx/" + s.spdxDoc.DocumentName
+	if len(spdxIds) == 0 {
+		// If there is no relationship that describes the document, we take a best guess for the pUrl.
+		spdxIds = append(spdxIds, "pkg:guac/spdx/"+s.spdxDoc.DocumentName)
+		s.topLevelIsHeuristic = true
+	}
 
-	if purl != "" {
-		topPackage, err := asmhelpers.PurlToPkg(purl)
+	for _, id := range spdxIds {
+		// Check if we can find a pUrl from a package before constructing a default one.
+		pUrl := ""
+		for _, pac := range s.spdxDoc.Packages {
+			if string(pac.PackageSPDXIdentifier) == id {
+				for _, ext := range pac.PackageExternalReferences {
+					if ext.RefType == spdx_common.TypePackageManagerPURL {
+						pUrl = ext.Locator
+					}
+				}
+			}
+		}
+
+		// Construct a default pUrl if none exist for the id.
+		pUrl = "pkg:guac/spdx/" + id
```

Comment:
hmm looks like this should be before the for loop? Also, i think we can use [GuacPkgPurl](https://github.com/guacsec/guac/blob/main/pkg/assembler/helpers/purl.go#LL178C2-L178C2) here